src/vcStringFormat.cpp

```diff
@@ -186,3 +190,67 @@ const char *vcStringFormat(char *pBuffer, size_t bufLen, const char *pFormatStri
 {
   return vcStringFormat(pBuffer, bufLen, pFormatString, &pString, 1);
 }
+
+// ****************************************************************************
+// Author: Damian Madden, February 2020
+const char *vcStringFormat_ImGuiClean(const char *pStr)
```

Comment:
This function name doesn't really make any sense?

Reply:
Yes I suppose a better fix would be to change udFilename, but solving it on either end works, and this fix is needed to for the other issue linked, the name implies that it makes a string clean for imgui

Reply:
> Yes I suppose a better fix would be to change udFilename, but solving it on either end works

Just to clarify, `udFilename` crashes when given a path with a `%` symbol in it?

> the name implies that it makes a string clean(sanitised) for imgui

I guess I wasn't very clear in why it didn't make any sense - the issues indicate that the problem is with `udFilename` and yet this function implies that it's a problem with ImGui, this doesn't make any sense.

Reply:
Yeah it crashes if passed a % symbol without a corresponding argument

Reply:
Then this should be fixed in `udFilename` instead of here - it looks like calls to `udFilename::SetFromFullPath` need to use `("%s", args)` instead of just `(args)`. Can you also add a unit test for this, call each Set function with a path that contains `%s` or something and expect the path to still be `%s`.

Reply:
"...but solving it on either end works, and this fix is needed to for the other issue linked", which crashes in imgui

Reply:
I'd think since as far as QA is concerned this solves both issues, then we could add one like "Fix udFiename handling % characters"

Reply:
> "...but solving it on either end works, and this fix is needed to for the other issue linked", which crashes in imgui

I'm not finding this crash in ImGui you're talking about, neither linked issue says anything about ImGui. I've patched `udFilename` and an oversight in the `vcWebFile` load function, and I'm not seeing any crashes in ImGui? I was assuming this was going to be the same problem, passing in the filename as a format string instead of as an argument, however, it doesn't seem to be an issue at all.

> I'd think since as far as QA is concerned this solves both issues, then we could add one like "Fix udFiename handling % characters"

Tricking QA into thinking something is fixed isn't how you should be approaching bug fixes.